### PR TITLE
Scheduler in a pod

### PR DESCRIPTION
**Pull request by ArtfulCoder**
#5011

### CODE REVIEW

**Comment by ArtfulCoder** [comment]
still running e2e tests

**Comment by ArtfulCoder** [comment]
manually tried scheduling a pod, it worked.
still waiting for e2e

**Comment by ArtfulCoder** [comment]
Ran 35 of 39 Specs in 1125.037 seconds
SUCCESS! -- 35 Passed | 0 Failed | 0 Pending | 4 Skipped I0414 13:32:49.415431   22543 driver.go:117] All tests pass

**Comment by dchen1107** [comment]
cc/ @justinsb on aws and @derekwaynecarr on vagrant too.

**Comment by derekwaynecarr** [comment]
@dchen1107 - thanks, I can test this early tomorrow and validate.

File: cluster/saltbase/salt/kube-master-addons/init.sls [hunk 1]
```diff
@@ -35,3 +35,4 @@ kube-master-addons:
     - restart: True
     - watch:
       - file: master-docker-image-tags
+      - file: /etc/kubernetes/kube-master-addons.sh
```
Comment (dchen1107) [inline cluster/saltbase/salt/kube-master-addons/init.sls]: Why we watch this script file here?

**Comment by dchen1107** [comment]
LGTM overall except one question.

File: cluster/saltbase/salt/kube-master-addons/init.sls (hunk 1 above)
Comment (ArtfulCoder) [inline cluster/saltbase/salt/kube-master-addons/init.sls]: @dchen1107 so that the service can restart if we have to load more docker images.

**Comment by dchen1107** [comment]
Can you rebase first to get this in?

**Comment by ArtfulCoder** [comment]
@dchen1107 rebased - thank you!

**Comment by dchen1107** [comment]
LGTM to me. Will merge on green. @derekwaynecarr @justinsb please note that this might break other cluster providers.

**Comment by ArtfulCoder** [comment]
kube-push works too.

**Comment by justinsb** [comment]
AWS is happy (as happy as ever): http://kubernetes-dashboard.meteor.com/

I haven't figured out how to easily e2e test PRs yet.  Frankly, I don't know if that will be worth the effort vs accepting that master might temporarily not pass all e2e tests on all platforms.  (One option might be to maintain a "premaster" integration branch, which is merged to master only when all e2e tests pass, but I don't think we feel that strongly about master passing 100% of the time)

**Comment by derekwaynecarr** [comment]
I will report and resolve an issue if I find any (don't think I will)

**Comment by dchen1107** [comment]
Great, thanks!

**Pull request merged**